src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -171,9 +171,13 @@ def convert_components_to_artifacts(
             # According to PEP-0589 all keys must be present in a TypedDict.
             # See https://peps.python.org/pep-0589/#totality
             item = DependencyInfo(
-                version=component.get("version") or "",
-                group=component.get("group") or "",
+                scheme="pkg",
+                type_="maven",
```

Comment:
The class isn't exactly language agnostic when it uses `groupId:artifactId` as the artifact keys, but I agree otherwise.

Reply:
@benmss Are you referring to [this line](https://github.com/oracle/macaron/pull/388/files/57d13c9d300bbe896b8c315670ef5accfc3e2333#diff-b2e47510ad053dfc5cd207b480a61ff1dce1c947acbc1b89b3a7a0e0b1ae84b6L170)?
```
            key = f"{component.get('group')}:{component.get('name')}"
```
From this particular line, it looks to me like this function seems to only support Java previously because a component that is not a Java package does not have the `group` attribute.


Suggestion: To make this function actually language agnostic, maybe instead of reading attributes such as `'group'` or `'name'` from CycloneDx SBOM Components, we should read the `'purl'` attribute instead ([example](https://github.com/oracle/macaron/blob/32f2f6d01763f867b7092faccc5e29b0970c0802/tests/dependency_analyzer/cyclonedx/resources/apache_maven_root_sbom.json#L98)). @benmss I think you should consider refactoring this function in this direction of utilizing the `'purl'` attribute and remove reads from attributes that do not seem language agnostic (e.g. `'group'`). Let me know what you think.

Additional info I found:
* [CycloneDX SBOM version 1.4 schema in protobuf](https://github.com/CycloneDX/specification/blob/cc15c8550e87dfa9ee051e6795e74a3c6c641264/schema/bom-1.4.proto#L101) (I assume Macaron is supporting version 1.4 because the example SBOMs in our test resources all follows version 1.4).

Reply:
Using `purl` could be a good candidate but because it is optional, the refactoring might not be simple. I have added an issue to refactor this function separately to support other languages: https://github.com/oracle/macaron/issues/464

@benmss you can add a TODO linking this issue but for now, it would be better to avoid fixing the `type`.

Reply:
As we agreed to retrieve the dependency type directly from the SBOM, the options appear to be:

1. Try to get the type from the optional PURL
2. Fix the type as "maven"